src/quality-checker.js

```diff
@@ -0,0 +1,116 @@
+const { spawn } = require('child_process');
+const fs = require('fs');
+const readline = require('readline');
+const path = require('path');
+const Papa = require('papaparse');
+
+module.exports = {checkCode, processCouplingMetric};
+
+const papaparseConfig = {
+    header: true,
+    skipEmptyLines: true
+};
+let problemsFound = [];
+
+/**
+ * Uses PMD to check the Java code in the sourceDirectory, finds all the problems from the rule set
+ * and outputs it into the JSON file at outputFileLocation for the front-end to read later,
+ * returns a promise to indicate when it's done or what errors there were
+ * @param {String} sourceDirectory reads code from here
+ * @param {String} outputFileLocation location of the output JSON file
+ * @returns {Promise}
+ */
+function checkCode(sourceDirectory, outputFileLocation) {
+    return new Promise((resolve, reject) => {
```

Comment:
I think we should keep problemsFound as a local variable because this is going to be called for each commit to check the code quality for each commit.

Reply:
I also want to mention that for the rule set, we may want to limit ourselves to a couple rules (eg. 1-3 rules from code style, 1-3 from best practices) in an effort to reduce overhead. Also, as we want to display these items, it's better that we target specific issues instead of trying to throw multiple darts at the board

Reply:
variable moved